Tests/tests/test_RDKShell.cpp

```diff
@@ -0,0 +1,164 @@
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("removeKeyListener")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("removeKeyMetadataListener")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("resetInactivityTime")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("resumeApplication")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("scaleToFit")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setBounds")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setCursorSize")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setFocus")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setHolePunch")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setInactivityInterval")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setLogLevel")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setMemoryMonitor")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setOpacity")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setScale")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setScreenResolution")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setTopmost")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setVirtualResolution")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setVisibility")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setGraphicsFrameRate")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("showCursor")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("showFullScreenImage")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("showSplashLogo")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("showWatermark")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("suspend")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("suspendApplication")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("keyRepeatConfig")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setAVBlocked")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("getBlockedAVApplications")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("addEasterEggs")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("removeEasterEggs")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("enableEasterEggs")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("getEasterEggs")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("launchFactoryApp")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("launchFactoryAppShortcut")));
+    }
+
+TEST_F(RDKShellTest, visibility)
+{
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("setVisibility"), _T("{\"client\": \"org.rdk.Netflix\","
+                                                                                             "\"callsign\": \"org.rdk.Netflix\","
+                                                                                             "\"visible\": true}"), response));
+    EXPECT_EQ(response, string("{\"success\":true}"));
```

Comment:
a test case should actually test something. how do you verify the business logic of "setVisibility"? i.e. how do you verify it works correctly instead of returning `"{\"success\":true}"`. by looking at this test case I assume "setVisibility" may just return `"{\"success\":true}"` and that will be OK for your test suite? if you are verifying that "setVisibility" triggers the backend then please add expectations for that. this concerns other test cases as well